src/store/auth.ts

```diff
@@ -0,0 +1,16 @@
+import { UserResponse } from 'api/auth/model';
+import { create } from 'zustand';
+
+interface AuthStore {
+  user: UserResponse | null;
+  setUser: (auth: UserResponse) => void;
```

Comment:
User라는 타입을 따로 만들어 사용하는건 어떨까요?
다른 쪽에서 해당 타입을 사용하다면 신경써야하니.. 같은 점을 생각해봤어요
자동로그인에서 에러가 난다면 `setUser`에 `null`이 들어갈 수 있으니 해당 타입도 짧게 쓸 수 있을 것 같네요!
```ts
export const User = z.nullable(UserResponse);
```
```suggestion
  user: User;
  setUser: (auth: User) => void;
```

Reply:
오... 확실히 더 편하게 사용할수있을것같네요..! 
타입 이름이 애매하긴한데, 개인적으로 그냥 `User`이라는 말이 모호한것같아 `NotNullUser` 이런식으로 지정해도되려나.. 

뭔가 null 이 허용되지않는 타입이라는 느낌을 주고싶은데, `RequiredUser`이런식으로 하는건 어떨까요? 좀 어색한 감이 있을까요.. ㅋㅋㅋ

Reply:
아예 null값 예외처리를 스키마를 이용해서 처리하는것도 좋은 방법이네요
위와같이 User라는 스키마를 정의하면 타입으로 또 따로 정의해야해서,
이름도 바꿔서  `NullableUser` 타입으로 수정했습니다

Reply:
ㅋㅋㅋㅋㅋ 가독성 측면에서 "유저" 라는 타입을 정의해야 한다고 생각하기에, (재사용할일이 많을거예요) `type User`를 정의할 필요가 있다고 생각했어요.

User는 스토어의 관점에서는 null이거나, UserResponse를 담은 객체일테고,
사용하는 입장에서는 `initialState`가 null 인 UserResponse를 담을 객체이니,
그냥 UserResponse 자체를 담거나, nullable을 추가해서 써도 괜찮겠다 정도의 의도였어요.

아마 NullableUser라고 이름지으면 추후에 유저 타입을 활용할 필요가 생겨 IDE에서 검색하게 된다면 `Use...` 이런식으로 찾게 될텐데 활용하지 못할 것 같아요 ㅋㅋㅋ

그리고 수정된 부분이 PascalCase에서 벗어났어요!

Reply:
위에 model에 관한 파일 구조를 한번 더 수정하면서, 애매하던 부분이 해결된것같습니다... ㅎㅎ
store의 관점에서는 nullable하니 그냥 `User`로 정의해될것같고, 활용도 더 좋을것같네요!